AppLovin MAX Demo App - Java/app/src/main/java/com/applovin/enterprise/apps/demoapp/data/main/FooterType.java

```diff
@@ -0,0 +1,33 @@
+package com.applovin.enterprise.apps.demoapp.data.main;
+
+import android.os.Build;
+
+import com.applovin.enterprise.apps.demoapp.BuildConfig;
+import com.applovin.sdk.AppLovinSdk;
+
+public class FooterType
+        implements ListItem
+{
+
+    /**
+     * @return The device details: App Version, SDK Version, OS Version
+     */
+    public String getAppDetails()
+    {
+        String appVersion = BuildConfig.VERSION_NAME;
```

Comment:
This and `sdkVersion` should be final as well.

Reply:
Do we use a linter to help me catch cases like this in Android Studio?

Reply:
I don't think there is. Also, we have recently started dropping `final` keyword from all local variables, unless required. So, should probably drop final from all the parameters.

Reply:
I don't know if the linter will catch it. It's more for when we read it; it's easy to tell if the variable shouldn't be changed. 

You can think of it, as whenever you use `val` in kotlin you use `final` in Java.

Reply:
> I don't think there is. Also, we have recently started dropping `final` keyword from all local variables, unless required. So, should probably drop final from all the parameters here.

what do you mean by unless required?

Reply:
When the variable is used by an anonymous class, like a `Runnable` or a Callback requiring the variable to be marked final. When the variable will be used outside the context of where its declared.

Reply:
sounds good, will remove final in this case.